### PR TITLE
Small fixes in the Readme

### PR DESCRIPTION
**Pull request by 0xc0170**
@c1728p9 @ashok-rao Please review

### CODE REVIEW

**Review by ashok-rao — APPROVED** [review]
LGTM 👍.. Apart from the mention of the blinky_application.bin in main.cpp in the repo, other docs changes are fine. Will @c1728p9 be addressing the point in main.cpp?

**Comment by 0xc0170** [comment]
> Apart from the mention of the blinky_application.bin in main.cpp in the repo, other docs changes are fine. Will @c1728p9 be addressing the point in main.cpp?

This define: https://github.com/ARMmbed/mbed-os-example-bootloader/blob/master/main.cpp#L5 ?

**Comment by ashok-rao** [comment]
Yes @0xc0170 , that one.. SD card for just building the bootloader binary (as per the README) seems a bit out-of-place..also we might want to make this as starting point for all examples..even filesystems ones.. so better to keep it generic than hard code a particular binary / example to it..

@c1728p9 @screamerbg , @sg- .. what do you think?

**Comment by 0xc0170** [comment]
> Yes @0xc0170 , that one.. SD card for just building the bootloader binary (as per the README) seems a bit out-of-place..also we might want to make this as starting point for all examples..even filesystems ones.. so better to keep it generic than hard code a particular binary / example to it..

New update - removed not needed parts (CI, SD card), replace nucleo target by a generic target name

**Comment by ashok-rao** [comment]
LGTM .. we just need @c1728p9 to confirm about https://github.com/ARMmbed/mbed-os-example-bootloader/blob/master/main.cpp#L5 .. Rest all good to go!

**Comment by sg-** [comment]
I've always liked the idea of offset built images having the offset in the name (see DAPLink). I think we should implement that here / in the tools too rather than using the word bootloader everywhere

**Comment by 0xc0170** [comment]
> I've always liked the idea of offset built images having the offset in the name (see DAPLink). I think we should implement that here / in the tools too rather than using the word bootloader everywhere

sounds good to me to have it in the name. @c1728p9 What do you think? If we make it in the tools, I can update the docs

**Comment by ashok-rao** [comment]
That would avoid a LOT of confusions.. +1 to have the distinctions in the file names itself.

File: README.md [hunk 1]
```diff
@@ -2,29 +2,29 @@
 
 This example shows how to create a bootloader. For steps on how to create an application which uses this bootloader, see [mbed-os-example-bootloader-blinky](https://github.com/ARMmbed/mbed-os-example-bootloader-blinky).
 
+To read more about the bootloader, please visit [bootloader tutorial](https://docs.mbed.com/docs/mbed-os-handbook/en/latest/advanced/bootloader/).
+
 ## Required hardware
 * A supported board - [u-blox EVK-ODIN-W2](https://developer.mbed.org/platforms/ublox-EVK-ODIN-W2/), [Nucleo F429ZI](https://developer.mbed.org/platforms/ST-Nucleo-F429ZI/) or [K64F](https://developer.mbed.org/platforms/FRDM-K64F/).
-* CI Test shield.
-* SD card.
```
Comment (c1728p9) [inline README.md]: The bootloader in the example is setup specifically to use an SD card on the CI Test shield. Without this none of the bootloader functionality can be used.

**Comment by c1728p9** [comment]
I don't have strong feelings either way on having the starting address of an application/bootloader as part of the name. @theotherjimmy what are your thoughts on this?

File: README.md (hunk 1 above)
Comment (0xc0170) [reply README.md]: Will fix

Comment (ashok-rao) [reply README.md]: @c1728p9 , as per the README, this example only "builds" the bootloader binary.. there is no mention of updating it or any other BL functionality. Should we modify the README then?

**Comment by ashok-rao** [comment]
>I don't have strong feelings either way on having the starting address of an application/bootloader as part of the name. @theotherjimmy what are your thoughts on this?

I think this refers to using the "_application" or the "_bl" in the file names? And not the actual offset addresses..I maybe wrong..

File: README.md [hunk 2]
```diff
@@ -72,6 +72,8 @@ Total Flash memory (text + data + misc): 60554 bytes
 Image: .\BUILD\NUCLEO_F429ZI\ARM\mbed-os-example-bootloader.bin
 ```
 
+It creates two binary files. The original uncombined image is in the output directory: <project-name>_application.bin and the bootloader image is <project-name>.bin.
+
 ## Next steps
 
 When the build succeeds, you have created a bootloader for your target. The next step is to build an application you can combine with your bootloader to create a loadable image. You can find a blinky application that uses this bootloader in the [mbed-os-example-bootloader-blinky](https://github.com/ARMmbed/mbed-os-example-bootloader-blinky) project.
```
Comment (ashok-rao) [inline README.md]: "When it builds, you have created a bootloader for your target. The next step is to build an application .... " I think this statement is ambiguous. So, this application "only" builds the BL binary. I have to go to another application that uses this bootloader. So, why do we need to mention the SD card or test the "update" functionality of the bootloader here with an  SD card??

**Review by ashok-rao — CHANGES_REQUESTED** [review]
Some lines in the "Next steps" section are confusing. Please review.

How about finding some other means of testing the BL -> like flash the new bootloader, make sure the DAPLink interface can still be upgraded/downgraded on the target or atleast make sure the DAPLink can be written to the device again & the target is still enumerated as mbed ? etc etc..

**Comment by sg-** [comment]
The proposal would be the following:
- application without a bootloader, <program name>
- application with constrain, <program name>-<?? or maybe blank ??>
- application with bootloader.bin, <program name>-<loadaddress 0x123455>

File: README.md (hunk 1 above)
Comment (0xc0170) [reply README.md]: I'll fix it

File: README.md (hunk 2 above)
Comment (0xc0170) [reply README.md]: I'll rewrite this section, noticed the same

**Comment by 0xc0170** [comment]
@ashok-rao I added SD card explanation, please review

**Comment by 0xc0170** [comment]
@ashok-rao Please review, or make edits here. I would like to proceed with this Readme update.

**Comment by ashok-rao** [comment]
@0xc0170 .. Ok, let's proceed with this for now. I'll make edits as I find suitable and raise other PR's. Thanks!

**Review by ashok-rao — APPROVED** [review]

**Comment by ashok-rao** [comment]
I have approved now. Please go ahead to merge :-)

**Pull request merged**